attachment.go

```diff
@@ -0,0 +1,147 @@
+package confluence
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+)
+
+// NOTE: The `_links.download` key is not documented in the Confluence REST API Reference. However,
+//       it is referenced here: https://community.atlassian.com/t5/Answers-Developer-Questions/confluence-pages-get-attached-files-via-REST-API/qaq-p/529873#M67529
+type Attachment struct {
+	ID string `json:"id,omitempty"`
+	Title string `json:"title,omitempty"`
+	Links struct {
+		Download string `json:"download,omitempty"`
+	} `json:"_links,omitempty"`
+}
+
+type AttachmentResults struct {
+	ResultPagination
+	Results []Attachment `json:"results"`
+}
+
+func (w *Wiki) getAttachmentByFilenameEndpoint(contentID string, filename string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment?filename=%v", w.endPoint.String(), contentID, filename))
+}
+
+func (w *Wiki) createAttachmentEndpoint(contentID string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment", w.endPoint.String(), contentID))
+}
+
+func (w *Wiki) updateAttachmentEndpoint(contentID string, attachmentID string) (*url.URL, error) {
+	return url.ParseRequestURI(fmt.Sprintf("%v/content/%v/child/attachment/%v/data", w.endPoint.String(), contentID, attachmentID))
```

Comment:
I would encourage you to use the more explicit verbs available with `fmt.Sprintf()`. Since these are all strings, `%s` seems appropriate.

https://gobyexample.com/string-formatting

Reply:
- [ ] use more appropriate `%s` verb